src/nvim/msgpack_rpc/channel.c

```diff
@@ -313,34 +312,35 @@ static void handle_request(Channel *channel, msgpack_object *request)
     return;
   }
 
-  // Retrieve the request handler
   MsgpackRpcRequestHandler handler;
-  Array args = ARRAY_DICT_INIT;
+
   msgpack_object *method = msgpack_rpc_method(request);
+  handler = msgpack_rpc_get_handler_for(method->via.bin.ptr,
+                                        method->via.bin.size);
+
+  // check method
+  if (handler.fn == msgpack_rpc_handle_missing_method) {
+    const char *m = method->via.bin.size > 0 ?
```

Comment:
Is the method name always NUL-terminated? The function `api_set_error()` uses format string `"%s"` to copy the method name.

Do we want to explicitly handle an empty string that contains a NUL-byte and has size 1 ("" or `"<empty>"`)?

Reply:
Would it be the safe bet to return to the string?